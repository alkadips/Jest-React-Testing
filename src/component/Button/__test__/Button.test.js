import React from "react";
import ReactDOM from 'react-dom';
import Button from "./../Button";
import{render}from '@testing-library/react';
import renderer from 'react-test-renderer';



it("renders without crashing",()=>{
const div=document.createElement("div");
ReactDOM.render(<Button></Button>,div)
});
it("renders button correctally",()=>{
   const {getByTestId}= render(<Button label="click me please"/>)
   expect(getByTestId('button')).toHaveTextContent("click me please")
});

it("renders button correctally",()=>{
    const {getByTestId}= render(<Button label="save"/>)
    expect(getByTestId('button')).toHaveTextContent("save")
 });
 it("matches snapshot",()=>{
     const tree=renderer.create(<Button label="save"></Button>).toJSON();
     expect(tree).toMatchSnapshot();
 })