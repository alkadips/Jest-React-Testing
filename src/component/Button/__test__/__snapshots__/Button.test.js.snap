// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="button-css"
  data-testid="button"
>
  Hi this is alka
  save
</div>
`;
